### DISCUSSION TITLE
Phrozen Sonic Mini 8k  ( firmware & multiple exposure )

### DISCUSSION THREAD

**Discussion by Hazoooka** [sn4k3/UVtools#894] [General]
Hi all.

For those using this printer for whatever form of multiple exposures.
I have just confirmed that the latest firmware update V466-1.6 is not reading additional exposures WaitBeforePrint times.
As a result, instead of having subsequent exposures within a fraction of the second after each other it’s reading the main layer WaitBeforePrint time making the print longer analogically to the number of additional exposures.
In other words, 2 exposures will double your print time, 3 will triple it.

Firmware V453-1.5 still reads this value and makes the print in the reasonable time.

So for you using multiple exposures, if you want to keep the print times shorter do not upgrade to V466-1.6.
Stay on V453-1.5.

I have tested this behaviour both by settings in Phased Exposures and Double Exposures in 4.0.4, as well as by editing layers values with Tools > Edit print parameters.

[edit]
The newest firmware supports PRZ format in Mini 8k, but it does not change the problem.
Both CTB and PRZ do not support WaitBeforePrint changes.

Regards,
Witold

**Comment by sn4k3** [comment]
If current firmware does "everything" you wish and it's compatible with most stuff **NEVER** upgrade firmware.
This behavior is not new, instead fix things or going for the better is the otherwise, other users faced same thing in other machines.

Please DON'T upgrade firmware if machine behaves as expected.
Currently the mSLA ecosystem is to screw the costumer and lock it more.

PS: Double Exposures can be still used on newest releases via Terminal and invoking the dialog.

**Comment by Hazoooka** [comment]
As far as I have analysed the code, and I am not a programmer, it looks to me that Phased Exposures base on Double Exposures still present in the latest builds.
Is it also possible to use Phased Exposures from Terminal?
I presume that invoiking a dialog means writing the script, correct?

**Reply by sn4k3** [reply]
Invoking dialog from UI requires only 1 line of code. `File - Terminal` and send the command:

```C#
ShowRunOperation(new OperationDoubleExposure())
```

**Comment by Hazoooka** [comment]
Really cool thing is that options menu can be opened by drag&drop the options file ( .uvtop )

**Reply by sn4k3** [reply]
Yes, that's also an option if you have the uvtop file. They can even run directly without show the prompt.

**Comment by Hazoooka** [comment]
Plus they can be edited externally, so we can open the options menu pre-filled by the uvtop settings.
In most cases only small adjustments are required and GO
